src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -299,6 +296,21 @@ private function getDenormalizer($data, string $class, ?string $format, array $c
         return null;
     }
 
+    private function hasScalarDenormalizer(): bool
+    {
+        if (null !== $this->hasScalarDenormalizer) {
+            return $this->hasScalarDenormalizer;
+        }
+
+        foreach ($this->normalizers as $normalizer) {
+            if ($normalizer instanceof ScalarDenormalizer) {
```

Comment:
What about using `supportsNormalization` instead?

This may fail if someone decorated the normalizer (e.g for tracking purposes).

Reply:
Indeed, checking the type was a bid idea just to avoid some possibly costly computation in a custom normalizer but it does more wrong than good.
I think I will go back to my first idea and use the `Seriliazer::getDenormalizer()` method instead since it's already cached.